wordpress-mobile/AztecEditor-Android
# Added support for the style attribute interpretation of CSS

### Fix
Added new functionality that translates the color CSS Style property of the inline HTML style attribute. This feature covers these types of scenarios:

**Basic** 
`<b style="color:red">Red Bold</b>`

**Nested - Color style applied to a single nested child**
```
<ul>
    <li style=“color:#ff00ff00”>Red List Item</li>
    <li>Non-styled List Item</li>
</ul>
```
**Nested - Color style applied to full element including children**
```
<ul style=“color:lime”>
    <li>I’m citrusy</li>
    <li>I’m green with envy</li>
</ul>
```
**Accepted Color Values**
* Named color strings (ex. “blue”, “red”, “teal”)
    * ColorConverter holds a static list of CSS Color translations
    * If color not found in map, android.graphics.Color.parseColor() is used
* Android color resources (ex. `“@black”`)

**Invalid or Empty Color Values**
If no translation match is found for the color string, the converter will return ColorConverter.COLOR_NOT_FOUND. At this point this color style property will be ignored. 

**Design Overview**
- The main class for parsing is the **InlineCssStyleFormatter**. This can be expanded to cover other css style properties in the future. 
- Style attributes are applied in two places and only to spans that implement _IAztecAttributedSpan_
    * Html.java in the end(..) method. The only tags styled here are the ones not handled by the AztecTagPlugin and any additionally provided IAztecPlugin.
    * AztecTagPlugin in the end(…) method. 
- **IAztecAttributedSpan** now has an extension function applyInlineStyleAttributes(…) for ease of applying styles when working in Kotlin.
- **_ColorConverter_** is a utility for translating named colors into color ints.

**Automated Tests**
The following tests have been created for this feature:
- **ColorConverterTest** - tests parsing valid and invalid color values.
- **HtmlAttributeStyleColorTest** - verifies attributes are parsed and applied to a sampling of spreads that cover the bases for each group:
    * Single List Item
    * Whole List
    * Basic tag processed via the Html class
    * Basic tag processed via the AztecTagHandler

**Actively seeking feedback (aka: a cry for help)**
The color style formatting for **AztecURLSpan** and **AztecCodeSpan** are overwritten by AztecText during switchToAztecStyle(). I’m not sure how to address this or if this is a concern, but I just wanted to mention it just in case 😃

### Test
1. Switch to the HTML source view.
2. Enter `<b style=“color:red”>This text is red and bold</b>`
3. Switch back to the Editor view.
4. The text should be displayed in the color red.

### Review
@0nko
